### PR TITLE
Sharezone Wrapped

### LINKED ISSUE #1605: Sharezone Rewrapped

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Description

This PR adds the Sharezone Wrapped feature. It is a test for this year if users want this feature and if it's worth maintaining. Therefore, the code is not the best.

Users can access the page via the settings. The tile on the settings page is behind a feature flag. Therefore, we can disable this tile when the last state has started with the new school year.

Also, users will get a tip card on the dashboard page to view the wrapped.

## Demo

image

image

https://github.com/SharezoneApp/sharezone-app/assets/24459435/0e23f07c-a708-457a-95a8-366fe5b3962d

_Note: I renamed it from Rewrapped to Wrapped because I noticed Spotify calls it Wrapped: https://pr-newsroom-wp.appspot.com/2023-wrapped/_

Closes #1605

### CODE REVIEW

File: app/lib/sharezone_wrapped/sharezone_wrapped_page.dart [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                    const SizedBox(height: 12),
+                    ElevatedButton.icon(
+                      icon: const Icon(Icons.share),
+                      onPressed: () async {
+                        // final capture = await _screenshotController.capture();
+
+                        // setState(() {
+                        //   _imageData = capture;
+                        //   print("Saved!");
+                        // });
```
Comment (nilsreichardt) [inline app/lib/sharezone_wrapped/sharezone_wrapped_page.dart:196]: TODO

**Comment by Sharezone-Admin** [comment]
⬇️ Builds generated by [Codemagic](https://codemagic.io) for commit 63c716bdcf0bbc0d24a967e29b7a0d6b1618b986 ⬇️

_Note_: Only Sharezone team members are able to install the iOS app.

| iOS | | Android |
|:-:|:-:|:-:|
| iOS QR code <br /> [Download link](https://api.codemagic.io/artifacts/.eJwVwcuygiAAANB_ad8MGXpt0SKTNClTSDM2DSp0Cx_XB72-vrnnTK6rf05dXkBgzx8tcW9m9lJusLvehcdv_BOv5KWv1aYtzu9W4X20VcDLxhf1kxCf7p7klTXPpWabwU4b0Je7c76aJzPYaxByYV_XC1UcdUIiTV1d84Nu6aIvid3DoDxCwyFj_osycUxnBdbkZ4CsGbrOIE8sTa-yuWRs_cQpRiAtI_fTIWHmUxz-dZ01ZGahp9KoIGns2DdiFDG_Io9WuwxYOdpNaQEOLB0zriKgtHyKPfVRTWWTPmB9GZMCbo2gUn0zjBl1yflkdlKgIHQ2zuFmiPcrXi4nXxV3Yug.I9JI4L9eUzxGcS1_2fxgN_Je3Y0) | | Android QR code <br /> [Download link](https://api.codemagic.io/artifacts/.eJwVxktygjAAANC7uGdGgk1g4SJUiYCVQooWNhkEA2j4dSLEnL7tW71Vjf-5XcXWAdPKDTiAwvhJcLbz9IwyvzfJ-D1QWn3ewoUM88BcHyvmVdHJi6QSClNkNPFVpt7YlD14Ou0AN_dLY_j--7l-xrsjFSVvyp1lCGYl4VVPzpSaMvhMR4U258U-cAuGRfFGbFNQzQNcSAN2tg5ky_nmBVrwoGmxZAlI2fWRzBDvCUm-QHse7ub-xofu2Mu_T68ejBr6zOmdKY4QNcn68vA-kg4dcieI7dzNCQLwXpQ46iPXWl_aoppPytXQcyxxkKx5aiTVsl_yTHWLAGFdb7erX2mzYi0.N5oX-TvOLaJVzM59D1YX-eBRqZ0) |

<sub>(expires Tue, 23 Jul 2024 10:46:33 GMT)<br/>Codemagic build: [6677f98da51e38e291d64b44](https://codemagic.io/app/629c82ea463af7ff553fc7a5/build/6677f98da51e38e291d64b44); comment generated with [Codemagic App Preview](https://github.com/nilsreichardt/codemagic-app-preview)</sub>

**Comment by Jonas-Sander** [comment]
Just some notes before reviewing it fully:

If e.g. "Prüfungen" is 0 there is still the "1.", "2.", "3." subject list which doesn't really makes sense

I fear that if not all exams are added to Sharezone and the number of it is really off, then people might have some slight bias against posting a screenshot as to not have questions about the wrong number of exams. Thats of course just my imagination, might be completely different in reality. I'm also not sure what would be the best way to fix that. Maybe being able to hide the stat? But might be too much hassle for this first MVP version of wrapped or it might be non-obvious too the user. Or the problem doesn't really exist, idk... 🤷‍♂️

Ideally I would like a small hint that explains that the hours of lessons is based on the current lesson plan, we can skip that though. Also not sure if it would look good.

**Review by Jonas-Sander — APPROVED** [review]
Nice, good job! LGTM

**Pull request merged**